### PR TITLE
Feature/search mediasite api on sis course

### PR DESCRIPTION
**Pull request by bermudezjd**
This PR modifies the `search_term` logic when making Mediasite API calls that determine whether a given folder or catalog exists.  Prior to this change, the catalog or folder name was used as the primary basis for searching on catalogs and folders respectively.  GIven the oData quirks noted as part of this PR in the corresponding methods, the logic has been updated to use the (formerly named) alternative search term as the primary search term over the name if it's given.  In the case of course folder or catalog creation, the course sis id is part of the name, so searching on this id instead of the full name should provide more consistent search results.

Also included is some minor PEP-8 cleanup, as well as a comment about how the API is used when folder permissions are established.

@elliottyates 
@sapnamysore

### CODE REVIEW

File: mediasite/apimethods.py [hunk 1]
```diff
@@ -71,34 +72,37 @@ class MediasiteAPI:
...
     @staticmethod
-    def get_folder(name, parent_folder_id, alternative_search_term=None):
-        folders = MediasiteAPI.get_folders(name, parent_folder_id)
+    def get_folder(name, parent_folder_id, search_term=None):
+        # Search on the name being passed in, unless a search_term is provided
+        if search_term is None:
+            search_term = name
+
+        folders = MediasiteAPI.get_folders(search_term, parent_folder_id)
         folder = next((f for f in folders if f.Name == name), None)
```
Comment (elliottyates) [inline mediasite/apimethods.py]: Do we care if there are multiple folders matched? Should we log this, perhaps?

File: mediasite/apimethods.py [hunk 2]
```diff
@@ -71,34 +72,37 @@ class MediasiteAPI:
...
-        # ideally the course sis id
-        # if not folder and alternative_search_term is not None:
-        #     folders = MediasiteAPI.get_folders(alternative_search_term, parent_folder_id)
-        #     folder = next((f for f in folders if f.Name == name), None)
         return folder
 
     @staticmethod
     def get_folders(name, parent_folder_id):
+        """ The mediasite API doesn't respect the oData "eq" filter as an exact match.  Rather, it appears from testing that
+        it behaves more like a "contains".  In addition, while the oData docs desribe "eq" as being case sensitive, it appears
```
Comment (elliottyates) [inline mediasite/apimethods.py]: This is actually explicitly called out in the Mediasite API documentation, FYI -- see [section 8 of the preamble](https://mediasite.video.harvard.edu/mediasite/api/v1/$metadata#top). It doesn't mention all the quirks listed here though, so this is still helpful!

File: mediasite/apimethods.py [hunk 3]
```diff
@@ -139,29 +143,30 @@ def get_or_create_folder(name, parent_folder_id, alternate_search_term=None, is_
...
-        catalogs = MediasiteAPI.get_catalogs(name)
+    def get_catalog(name, course_folder_id, search_term=None):
+        """ See oDAta note above in `get_folders` method """
+
+        # Search on the name being passed in, unless a search_term is provided
+        if search_term is None:
+            search_term = name
+
+        catalogs = MediasiteAPI.get_catalogs(search_term)
         catalog = next((c for c in catalogs if c.LinkedFolderId == course_folder_id), None)
```
Comment (elliottyates) [inline mediasite/apimethods.py]: Likewise, do we care if we match on multiple catalogs?

**Comment by elliottyates** [comment]
👍 this actually will fix the issue @sapnamysore found in #7, where unicode folder names are created as expected but searching for them using an escaped string doesn't work. I'll still work on fixing that bug, but it might be sidestepped so that we can at least support creating unicode folder and catalog names seeing as we have this workaround to match based on course instance ID instead of title.

File: mediasite/apimethods.py (hunk 1 above)
Comment (bermudezjd) [inline mediasite/apimethods.py]: I think the multiple folders is an expected case using the oData query when searching on name.  If we're using the sis id, then there should only be a single match.  This code doesn't assume that the search term is a single match though... so you're thinking that in general you'd want to log the `folders` that are being returned?

File: mediasite/apimethods.py (hunk 3 above)
Comment (bermudezjd) [inline mediasite/apimethods.py]: I think the same expectations apply here.  Maybe it makes more sense to log results in a `debug` mode at a higher level?

File: mediasite/apimethods.py (hunk 1 above)
Comment (elliottyates) [inline mediasite/apimethods.py]: I'm thinking that we've made an assumption, and should assert that it's true in some way -- in other words, if `len(folder) > 0` perhaps we should log and fail (the `len()` may make the whole `next()` construction less useful, but with only one result, it doesn't matter what you're doing for efficiency's sake).

This is outside the scope of the PR though, I suppose. Just something I noticed.

Comment (elliottyates) [inline mediasite/apimethods.py]: Yes I would say this is tech debt. Thanks!

File: mediasite/apimethods.py (hunk 3 above)
Comment (elliottyates) [inline mediasite/apimethods.py]: Sure, we could certainly do that. I see that we aren't logging anything really other than the calls we're making to the mediasite API (not our Canvas calls, for instance, and no results of any of the calls to either API), so it might be helpful to make the debug chattier, given the volume of tool use. But that could be for tech debt, as mentioned in that other comment thread.

File: mediasite/apimethods.py (hunk 1 above)
Comment (bermudezjd) [inline mediasite/apimethods.py]: Created https://jira.huit.harvard.edu/browse/TLT-2879 as a related tech debt item.
